sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageSharedKeyCredential.java

```diff
@@ -111,15 +129,16 @@ public String computeHmac256(final String stringToSign) {
         return StorageImplUtils.computeHMac256(accountKey, stringToSign);
     }
 
-    private String buildStringToSign(URL requestURL, String httpMethod, Map<String, String> headers) {
+    private String buildStringToSign(URL requestURL, String httpMethod, Map<String, String> headers,
+        boolean logStringToSign) {
```

Comment:
Why not accept the context itself instead of a boolean?

Reply:
The context in the pipeline policy isnt an AzureContext, its like a HttpPipelineContext so I'd need to do that weird conversion somewhere. Would you prefer I do it in the pipeline policy?

Reply:
`HttpPipelineContext` is the combination of `HttpRequest` and `Context` where the `Context` APIs are wrapped and hidden by the `HttpPipelineContext` APIs, in the end it should contain all information from the `Context`.